src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/mediapicker/mediapicker.controller.js

```diff
@@ -98,21 +98,30 @@ angular.module("umbraco")
                         gotoStartNode();
                     }
                 } else {
-                    //if a target is specified, go look it up - generally this target will just contain ids not the actual full
-                    //media object so we need to look it up
+                    // if a target is specified, go look it up - generally this target will just contain ids not the actual full
+                    // media object so we need to look it up
                     var id = $scope.target.udi ? $scope.target.udi : $scope.target.id;
-                    var altText = $scope.target.altText;
-                    entityResource.getById(id, "Media")
+
+                    // ID of a UDI or legacy int ID still could be null/undefinied here
+                    // As user may dragged in an image that has not been saved to media section yet
+                    if(id){
+                        entityResource.getById(id, "Media")
                         .then(function (node) {
-                            $scope.target = node;
+                            //$scope.target = node; //Not 100% sure why we override it here?!
```

Comment:
@warrenbuckley I think this code must be reintroduced, if we are not 100% sure why it is used.. This is maybe due to infinite editing, or when multiple users work on the same content